python/pyspark/sql/functions.py

```diff
@@ -135,6 +158,24 @@ def sparkPartitionId():
     return Column(sc._jvm.functions.sparkPartitionId())
 
 
+def struct(*cols):
+    """Creates a new struct column.
+
+    :param cols: list of column names (string) or list of :class:`Column` expressions
+        that are named or aliased.
+
+    >>> df.select(struct('age', 'name').alias("struct")).collect()
+    [Row(struct=[u'2', u'Alice']), Row(struct=[u'5', u'Bob'])]
+    >>> df.select(struct([df.age, df.name]).alias("struct")).collect()
+    [Row(struct=[u'2', u'Alice']), Row(struct=[u'5', u'Bob'])]
```

Comment:
I think there is a bug with struct type conversion in Python (see age becomes a string). Will file a separate jira to fix this one.